### PR TITLE
Have "choose_free_port" set it 0 so OS picks port

### PR DESCRIPTION
**Pull request by lucksus**
## PR summary

This deactivates the [previously added](https://github.com/holochain/holochain-rust/pull/1946) code that manually sweeps through ports to find a open one to bind an interface to and rather relies on the OS finding a free port when binding to port 0.

We can easily get the bound address from JSON-RPC's `Server` and print that out for try-o-rama to use.

The problem with the previous approach was that there still was a split-second in which we had to let go of the port in order to have the JSON-RPC server bind to it.

This hopefully eliminates the last `Address already in use` problems that create false negatives in stress tests.

## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

- [ ] if this is a code change that effects some consumer (e.g. zome developers) of holochain core,  then it has been added to [our between-release changelog](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md) with the format 

```markdown
- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)
```

## documentation

- [ ] this code has been documented according to our [docs checklist](https://hackmd.io/@freesig/Hk9AmKJNS)

### CODE REVIEW

File: crates/conductor_lib/src/conductor/base.rs [hunk 1]
```diff
@@ -1538,29 +1539,44 @@ fn run_interface(
...
-            || {
-                let r = HttpInterface::new(port).run(handler, kill_switch);
-                println!("{}", magic_port_binding_string(&interface_config.id, port));
-                r
-            },
-        ),
 
+    match interface_config.driver {
+        InterfaceDriver::Websocket { port } => {
+            let port = if interface_config.choose_free_port.unwrap_or(false) {
```
Comment (AshantiMutinta) [inline crates/conductor_lib/src/conductor/base.rs]: I was wondering if choose_free_port needs to be an option? I'm thinking it is either you choose the free port or you don't right?

Comment (willemolding) [reply crates/conductor_lib/src/conductor/base.rs]: Its nicer if its an option right because then Serde can still deserialize it if the field isn't there?

Comment (AshantiMutinta) [reply crates/conductor_lib/src/conductor/base.rs]: We could use #[serde(default)] that way it default to false already

File: crates/conductor_lib/src/conductor/base.rs [hunk 2]
```diff
@@ -1517,6 +1517,7 @@ fn _make_interface(interface_config: &InterfaceConfiguration) -> Box<dyn Interfa
     }
 }
 
+#[allow(dead_code)]
 fn with_port_heuristic<T, F: FnOnce() -> T>(
```
Comment (AshantiMutinta) [inline crates/conductor_lib/src/conductor/base.rs]: Planning to use this in function in the future I guess? :)

File: crates/conductor_lib/src/conductor/base.rs (hunk 1 above)
Comment (willemolding) [reply crates/conductor_lib/src/conductor/base.rs]: oh yeah nice

**Pull request merged**